File: app/src/main/java/com/possible/architecturesample/di/ActivityComponent.java
package com.possible.architecturesample.di;

import android.app.Activity;

import com.possible.architecturesample.ui.MainActivity;

import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(MainActivity activity);

    Activity activity();
}

File: app/src/main/java/com/possible/architecturesample/di/ActivityComponent.kt
package com.possible.architecturesample.di

import android.app.Activity

import com.possible.architecturesample.ui.MainActivity

import dagger.Component

@PerActivity
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun inject(activity: MainActivity)

    fun activity(): Activity
}
